libgit/repo.go

```diff
@@ -262,3 +266,52 @@ func CreateRepoAndID(
 	}
 	return id, err
 }
+
+// DeleteRepo "deletes" the given repo in the given TLF.  RIght now it
+// simply moves the repo out of the way to a special directory, to
+// allow any concurrent writers to finish their pushes without
+// triggering conflict resolution.  The caller is responsible for
+// syncing the FS and flushing the journal, if desired.  It expects
+// the `config` object to be unique during the lifetime of this call.
+func DeleteRepo(
+	ctx context.Context, config libkbfs.Config, tlfHandle *libkbfs.TlfHandle,
+	repoName string) error {
+	// Create a unique ID using the verifying key and the `config`
+	// object, which should be unique to each call in practice.
+	session, err := config.KBPKI().GetCurrentSession(ctx)
+	if err != nil {
+		return err
+	}
+	uniqID := fmt.Sprintf("%s-%p", session.VerifyingKey.String(), config)
```

Comment:
I don't love using the pointer value of `config` as the unique ID. Is there a reason not to use `kbfscrypto.RandRead`?

Reply:
It was easier, and it's guaranteed to be unique for this call.  Technically a random number isn't guaranteed to be unique.  Why don't you like it?

Reply:
I don't understand how a pointer is guaranteed to be unique.

Reply:
It only needs to be unique "during the lifetime of this call", as the function comment explains.  A new config is passed into each of these calls, and therefore any calls happening at the same time will have unique pointers.

Reply:
But the new config's pointer can be the same as the old one...

Reply:
Oh nvmind I follow, different configs _at the same time_

Reply:
Yeah that's what I was thinking, but you're right.  By including it in the name, it needs to be unique for all time, or at least until the next delete run happens.  I'll fix it soonish.